scale/docs/architecture/workspaces.rst

```diff
@@ -140,3 +144,64 @@ The NFS broker requires one additional field in its configuration:
 
     The *nfs_path* is a required string that specifies the remote NFS path to use for storing and retrieving the
     workspace files. It should be in the format *host:/path*.
+
+S3 Broker *(experimental)*
+------------------------------------------------------------------------------------------------------------------------
+
+The S3 broker references a storage location that exists as an S3 bucket in your AWS account. Please take note of the
+bucket name, which is typically of the form *my_name.domain.com* since bucket names must be globally unique
+(See `Bucket Restrictions`_). The bucket must be configured for read and/or write access through an appropriate IAM
+account (Identity and Access Management). Once the IAM account is created and granted permissions to the bucket, then
+the *ACCESS KEY ID* and *SECRET ACCESS KEY* can be generated and used with this broker (See `AWS Credentials`_). These
+tokens allow 3rd party software to access resources on behalf of the associated account.
+
+.. _Bucket Restrictions: http://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html
+.. _AWS Credentials: http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html
+
+**Security**
+
+A dedicated IAM account should be used rather than the root AWS account to limit the risk of damage if a leak were to
+occur and similarly the IAM account should be given the minimum possible permissions needed to work with the bucket. The
+access tokens should also be changed periodically to further protect against leaks.
+
+While this broker is in the experimental phase, the access tokens are currently stored in plain text within the Scale
+database and exposed via the REST interface. A future version will maintain these values using a more appropriate
+encrypted store service.
+
+Example S3 broker configuration:
+
+.. code-block:: javascript
+
+   {
+      "version": "1.0",
+      "broker": {
+         "type": "s3",
+         "bucket_name": "my_bucket.domain.com",
+         "credentials": {
+            "access_key_id": "AKIAIOSFODNN7EXAMPLE",
+            "secret_access_key": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+         }
+      }
+   }
+
+The S3 broker requires the following additional fields in its configuration:
+
```

Comment:
I took those keys from the AWS documentation below and you'll note they have "EXAMPLE" at the end of the values to help indicate they're fake:
http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html

My thought was that when I was new to AWS the access key id and secret access key were a little confusing, so seeing some values that look real helps people confirm they are doing things correctly rather than just putting in something like XXX.

I would expect these keys to not actually go into version control, but rather they would be provided using the upcoming Workspace UI, do you think I should make some sort of change here?